covalent_braket_plugin/braket.py

```diff
@@ -73,56 +76,160 @@ def __init__(self, status, repo):
         super().__init__(self.message)
 
 
-class BraketExecutor(BaseExecutor):
+class BraketExecutor(AWSExecutor):
     """AWS Braket Hybrid Jobs executor plugin class."""
 
     def __init__(
         self,
         credentials: str,
         profile: str,
         s3_bucket_name: str,
-        ecr_repo_name: str,
         braket_job_execution_role_name: str,
+        time_limit: int,
+        poll_freq: int,
+        ecr_repo_name: str,
         quantum_device: str,
         classical_device: str,
         storage: int,
-        time_limit: int,
-        poll_freq: int,
         **kwargs,
     ):
-        super().__init__(**kwargs)
 
-        self.credentials = credentials
-        self.profile = profile
-        self.s3_bucket_name = s3_bucket_name
+        super().__init__(
+            credentials_file=credentials,
+            profile=profile,
+            s3_bucket_name=s3_bucket_name,
+            execution_role=braket_job_execution_role_name,
+            time_limit=time_limit,
+            **kwargs,
+        )
+
         self.ecr_repo_name = ecr_repo_name
-        self.braket_job_execution_role_name = braket_job_execution_role_name
         self.quantum_device = quantum_device
         self.classical_device = classical_device
         self.storage = storage
-        self.time_limit = time_limit
         self.poll_freq = poll_freq
 
-    def run(self, function: Callable, args: List, kwargs: Dict):
+    async def _upload_task(
+        self, function: Callable, args: List, kwargs: Dict, upload_metadata: Dict
+    ):
+        """
+        Abstract method that uploads the pickled function to the remote cache.
+        """
+        image_tag = upload_metadata["image_tag"]
+        task_results_dir = upload_metadata["task_results_dir"]
+        result_filename = upload_metadata["result_filename"]
+
+        loop = asyncio.get_running_loop()
+
+        fut = loop.run_in_executor(
+            None,
+            self._package_and_upload,
```

Comment:
@cjao I think the idea for upload task is to simply upload the pickled function to the S3 bucket and not to build the docker image (as it is currently doing in  `_package_and_upload`) as we can use the prebuilt image that lives in the ECR repo https://us-east-1.console.aws.amazon.com/ecr/repositories/public/348041629502/covalent-executor-base?region=us-east-1 which can be referenced using it's public URI `public.ecr.aws/covalent/covalent-executor-base:latest`. The dockerfile for the prebuilt image is here for reference: https://github.com/AgnostiqHQ/covalent-aws-plugins/blob/develop/Dockerfile.